lib/proxy/log_buffer/buffer.rb

```diff
@@ -0,0 +1,83 @@
+require 'proxy/log_buffer/ring_buffer'
+
+module Proxy::LogBuffer
+
+  LogRecord = Struct.new(:serial, :timestamp, :level, :message)
```

Comment:
Additionally, you are not persisting the serial counter between restarts, dropping it's monotonicity property, which makes it less useful. Maybe consider using a Time.to_f as a timestamp, which will pretty much be unique per message.